index.js

```diff
@@ -68,8 +68,13 @@ const serverReadyPromise = new Promise((resolve) => {
     resolveServerReady = () => resolve();
 });
 
-app.on('child-window-created', function(parentId, childId, childOptions) {
-
+//Event either comes from the runtime or the core when registering an externalWindow.
+// Payload is a browserWindow id
+app.on('child-window-created', function(parentBwId, childBwId, childOptions) {
+    const parent = BrowserWindow.fromId(parentBwId);
+    const child = BrowserWindow.fromId(childBwId);
+    const parentId = parent.webContents.id;
```

Comment:
Do these need `undefined` checks?  `parent && parent.webContents && parent.webContents.id`?

Reply:
they may need a top level check for parent. That being said if it was ever undefined we'd be in a terrible place anyways so not sure what we should be doing if the check fails.